src/dsnotificationinterface.cpp

```diff
@@ -39,9 +40,21 @@ void CDSNotificationInterface::UpdatedBlockTip(const CBlockIndex *pindexNew, con
 
     // DIP0001 updates
 
+    bool fDIP0001ActiveAtTipTmp = fDIP0001ActiveAtTip;
     // Update global flags
     fDIP0001LockedInAtTip = (VersionBitsState(pindexNew, Params().GetConsensus(), Consensus::DEPLOYMENT_DIP0001, versionbitscache) == THRESHOLD_LOCKED_IN);
     fDIP0001ActiveAtTip = (VersionBitsState(pindexNew, Params().GetConsensus(), Consensus::DEPLOYMENT_DIP0001, versionbitscache) == THRESHOLD_ACTIVE);
+
+    // Update min fees only if activation changed and we are using default minRelayTxFee
+    if (fDIP0001ActiveAtTipTmp != fDIP0001ActiveAtTip) {
+        if (!mapArgs.count("-minrelaytxfee")) {
+            ::minRelayTxFee = CFeeRate(fDIP0001ActiveAtTip ? DEFAULT_LEGACY_MIN_RELAY_TX_FEE / 10 : DEFAULT_LEGACY_MIN_RELAY_TX_FEE);
```

Comment:
May be it is better to add separate DEFAULT_DIP001_MIN_RELAY_TX_FEE constant instead of DEFAULT_LEGACY_MIN_RELAY_TX_FEE / 10? The same with DEFAULT_LEGACY_TRANSACTION_MINFEE / 10.